crates/uv-resolver/src/resolver/urls.rs

```diff
@@ -1,36 +1,51 @@
 use rustc_hash::FxHashMap;
 use tracing::debug;
 
-use distribution_types::{RequirementSource, Verbatim};
+use distribution_types::{
+    ParsedArchiveUrl, ParsedGitUrl, ParsedLocalFileUrl, ParsedUrl, RequirementSource, Verbatim,
+    VerbatimParsedUrl,
+};
 use pep508_rs::{MarkerEnvironment, VerbatimUrl};
 use uv_distribution::is_same_reference;
+use uv_git::GitUrl;
 use uv_normalize::PackageName;
 
 use crate::{DependencyMode, Manifest, ResolveError};
 
 /// A map of package names to their associated, required URLs.
 #[derive(Debug, Default)]
-pub(crate) struct Urls(FxHashMap<PackageName, VerbatimUrl>);
+pub(crate) struct Urls(FxHashMap<PackageName, VerbatimParsedUrl>);
 
 impl Urls {
     pub(crate) fn from_manifest(
         manifest: &Manifest,
         markers: &MarkerEnvironment,
         dependencies: DependencyMode,
     ) -> Result<Self, ResolveError> {
-        let mut urls: FxHashMap<PackageName, VerbatimUrl> = FxHashMap::default();
+        let mut urls: FxHashMap<PackageName, VerbatimParsedUrl> = FxHashMap::default();
 
         // Add the editables themselves to the list of required URLs.
         for (editable, metadata, _) in &manifest.editables {
-            if let Some(previous) = urls.insert(metadata.name.clone(), editable.url.clone()) {
-                if !is_equal(&previous, &editable.url) {
-                    if is_same_reference(&previous, &editable.url) {
-                        debug!("Allowing {} as a variant of {previous}", editable.url);
+            let editable_url = VerbatimParsedUrl {
+                parsed_url: ParsedUrl::LocalFile(ParsedLocalFileUrl {
+                    url: editable.url.to_url(),
+                    path: editable.path.clone(),
+                    editable: true,
+                }),
+                verbatim: editable.url.clone(),
+            };
+            if let Some(previous) = urls.insert(metadata.name.clone(), editable_url.clone()) {
```

Comment:
The construction of `VerbatimParsedUrl` seems awkward here, just a lot of conversion. Will it be removed eventually?

Reply:
Yes, `LocalEditable` will switch away from `url: VerbatimUrl` to a `VerbatimParsedUrl`.